std/random.d

```diff
@@ -564,62 +570,134 @@ Parameters for the generator.
+    private State state = defaultState();
+    // NOTE: the above is a workaround to ensure
+    // backwards compatibility with the original
+    // implementation, which permitted implicit
+    // construction.  With `@disable this();`
+    // it would not be necessary.
+
+    /**
+       Constructs a MersenneTwisterEngine object.
+    */
     this(UIntType value) @safe pure nothrow @nogc
     {
-        seed(value);
+        this.seed(value);
     }
 
-/**
-   Seeds a MersenneTwisterEngine object.
-   Note:
-   This seed function gives 2^32 starting points. To allow the RNG to be started in any one of its
-   internal states use the seed overload taking an InputRange.
-*/
-    void seed()(UIntType value = defaultSeed) @safe pure nothrow @nogc
+    /**
+       Constructs a MersenneTwisterEngine object using a range
+       to seed the generator, which must have at least as many
+       elements as `data`.
+    */
+    this(T)(T range)
+        if (isInputRange!T && is(Unqual!(ElementType!T) == UIntType))
+    {
+        this.seed(range);
+    }
+
+    /**
+       Constructs a MersenneTwisterEngine object by
+       duplicating the state of an existing instance
+    */
+    this(ref typeof(this) that)
+    {
+        this.state = that.state;
+    }
+
+    /**
+       Generates the default initial state for a Mersenne
+       Twister; equivalent to the internal state obtained
+       by calling `seed(defaultSeed)`
+    */
+    private static State defaultState() @safe pure nothrow @nogc
+    {
+        if (!__ctfe) assert(false);
+        State mtState;
+        seedImpl(defaultSeed, mtState);
+        return mtState;
+    }
+
+    /**
+       (Re)seeds the generator
+    */
+    void seed(UIntType value) @safe pure nothrow @nogc
+    {
+        this.seedImpl(value, this.state);
+    }
+
+    /**
+       Implementation of the seeding mechanism, which
+       can be used with an arbitrary `State` instance
+    */
+    private static void seedImpl(UIntType value, ref State mtState)
     {
         static if (w == UIntType.sizeof * 8)
         {
-            mt[0] = value;
+            mtState.data[$-1] = value;
         }
         else
         {
             static assert(max + 1 > 0);
-            mt[0] = value % (max + 1);
-        }
-        for (mti = 1; mti < n; ++mti)
-        {
-            mt[mti] =
-                cast(UIntType)
-                (1812433253UL * (mt[mti-1] ^ (mt[mti-1] >> (w - 2))) + mti);
-            /* See Knuth TAOCP Vol2. 3rd Ed. P.106 for multiplier. */
-            /* In the previous versions, MSBs of the seed affect   */
-            /* only MSBs of the array mt[].                        */
-            /* 2002/01/09 modified by Makoto Matsumoto             */
-            //mt[mti] &= ResultType.max;
-            /* for >32 bit machines */
+            mtState.data[$-1] = value % (max + 1);
         }
-        popFront();
+        static if (is(UIntType == uint))
+            enum UIntType f = 1812433253;
```

Comment:
It's odd that this branch is marked as not covered by codecov.io as it should be used for `Mt19937`.